pkg/appgw/internaltypes.go

```diff
@@ -127,18 +134,15 @@ func generateFrontendPortName(port Port) string {
 }
 
 func generateListenerName(listenerID listenerIdentifier) string {
-	if listenerID.UsePrivateIP {
-		return formatPropName(fmt.Sprintf("%s%s-%v%v-privateip", agPrefix, prefixListener, formatHostname(listenerID.HostName), listenerID.FrontendPort))
-	}
-	return formatPropName(fmt.Sprintf("%s%s-%v%v", agPrefix, prefixListener, formatHostname(listenerID.HostName), listenerID.FrontendPort))
+	return formatPropName(fmt.Sprintf("%s%s-%s", agPrefix, prefixListener, utils.GetHashCode(listenerID)))
 }
 
 func generateURLPathMapName(listenerID listenerIdentifier) string {
-	return formatPropName(fmt.Sprintf("%s%s-%v%v", agPrefix, prefixPathMap, formatHostname(listenerID.HostName), listenerID.FrontendPort))
+	return formatPropName(fmt.Sprintf("%s%s-%s", agPrefix, prefixPathMap, utils.GetHashCode(listenerID)))
```

Comment:
I propose we preserve the old property names when the new `hostname-extension` annotation is not in use.

We could a) add a bool `listenerID.usingHostnameExtension==true` to `listenerID` struct and set it when the annotation is in use and b) we can use that boolean here:  return the old string when the annotation is not in use;  return the new string when annotation is in use?

Reply:
Resolving this as we discussed that geting hashcode for names seems more correct and would avoid any issues that will arise with special characters in hostname